src/fa.c

```diff
@@ -437,6 +437,10 @@ static struct state *add_state(struct fa *fa, int accept) {
     for (struct trans *t = (s)->trans;                                  \
          (t - (s)->trans) < (s)->tused;                                 \
          t++)
+#define for_each_const_trans(t, s)                                      \
+    for (const struct trans *t = (s)->trans;                            \
```

Comment:
Another way to do this is to just have `for_each_trans` but use `typeof ((s)->trans) t`; or would that break clang ?